flash/audio/classification/data.py

```diff
@@ -0,0 +1,223 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib
+from typing import Any, Callable, List, Optional, Sequence, Tuple, Union
+
+import torch
+from PIL import Image
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torchvision import transforms as T
+from torchvision.datasets import VisionDataset
+from torchvision.datasets.folder import has_file_allowed_extension, IMG_EXTENSIONS, make_dataset
+
+from flash.vision.classification.data import ImageClassificationDataPipeline, ImageClassificationData
+from flash.core.data.datamodule import DataModule
+from flash.core.data.utils import _contains_any_tensor
+
+
+def _pil_loader(path) -> Image:
+    # open path as file to avoid ResourceWarning (https://github.com/python-pillow/Pillow/issues/835)
+    with open(path, "rb") as f, Image.open(f) as img:
+        return img.convert("RGB")
+
+_default_train_transforms = T.Compose([
+    # T.RandomResizedCrop(224),
+    # T.RandomHorizontalFlip(),
+    T.ToTensor(),
+    T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+])
+
+_default_valid_transforms = T.Compose([
+    # T.Resize(256),
+    # T.CenterCrop(224),
+    T.ToTensor(),
+    T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+])
+
+# todo: torch.nn.modules.module.ModuleAttributeError: 'Resize' object has no attribute '_forward_hooks'
+# Find better fix and raised issue on torchvision.
+_default_valid_transforms.transforms[0]._forward_hooks = {}
+
+class SpectrogramClassificationDataPipeline(ImageClassificationDataPipeline):
+
+    def __init__(
+        self,
+        train_transform: Optional[Callable] = _default_train_transforms,
+        valid_transform: Optional[Callable] = _default_valid_transforms,
+        use_valid_transform: bool = True,
+        loader: Callable = _pil_loader
+    ):
+        super().__init__(train_transform,
+                       valid_transform,
+                       use_valid_transform,
+                       loader)
+
+class SpectrogramClassificationData(ImageClassificationData):
```

Comment:
I don't need to override those functions `from_filepaths` and `from_folders`.

Reply:
Yes we do because the default image transforms are not good for spectrograms.